react/features/polls/components/PollDialog.js

```diff
@@ -0,0 +1,363 @@
+                }
+            }
+        });
+    }
+
+    _removeChoice: (number) => void;
+
+    /**
+     * Removes an item from list.
+     *
+     * @param {string} id - ID of item in list.
+     * @returns {void}
+     */
+    _removeChoice(id: string) {
+        const filteredChoices = Object.assign({}, this.state.choices);
+
+        delete filteredChoices[id];
+
+        this.setState({
+            choices: filteredChoices
+        });
+    }
+
+    _choiceTextChange: (string, string) => void;
+
+    /**
+     * Item text change handler.
+     *
+     * @param {string} id - ID of the item with change.
+     * @param {string} text - New text.
+     * @returns {void}
+     */
+    _choiceTextChange(id: string, text: string) {
+        const choice = this.state.choices[id];
+
+        this.setState({
+            choices: {
+                ...this.state.choices,
+                [id]: {
+                    ...choice,
+                    text
+                }
+            }
+        });
+    }
+
+    _createNewPoll: () => void;
+
+    /**
+     * Click handler for creating a new poll.
+     *
+     * @returns {void}
+     */
+    _createNewPoll() {
+        const { question, choices, poll } = this.state;
+        const { dispatch } = this.props;
+        const uniqueChoices = getUniquePollChoices(choices);
+
+        if (!question.text.trim() || Object.keys(uniqueChoices).length < 2) {
+            return false;
+        }
+
+        Object.keys(uniqueChoices).forEach(x => {
+            poll.choices.push(x);
+        });
+
+        const payload = {
+            poll,
+            question,
+            choices: uniqueChoices
+        };
+
+        dispatch(startPoll(payload));
+
+        // Clear Object
+        const newState = this._clearState();
+
+        this.setState({
+            ...newState
+        });
+    }
+
+    _onQuestionTextChange: (Object) => void;
+
+    /**
+     * Update the question text in local state.
+     *
+     * @param {event} event - Keyboard event.
+     * @returns {void}
+     */
+    _onQuestionTextChange(event: Object) {
+        const text: string = event.target.value;
+
+        this.setState({
+            question: {
+                ...this.state.question,
+                text
+            }
+        });
+    }
+
+    _endPoll: () => void;
+
+    /**
+     * Dispatch action to end poll.
+     *
+     * @returns {boolean}
+     */
+    _endPoll() {
+        const { dispatch } = this.props;
+
+        dispatch(endPoll());
+    }
+
+
+    _vote: (string) => void;
+
+    /**
+     * Handle logic for voting for a specific option.
+     *
+     * @param {string} id - ID of selected item.
+     * @returns {boolean}
+     */
+    _vote(id: string) {
+        const { dispatch } = this.props;
+
+        dispatch(vote(id));
+    }
+
+    _clearState: () => Object;
+
+    /**
+     * Returns a new object that can be set to clear state.
+     *
+     * @returns {Object}
+     */
+    _clearState() {
```

Comment:
This isn't really "clearState". clearState to me sounds like a verb that is going to remove all state. This method is more of a createNewState or something like that.